### ISSUE TITLE
Parsing 3.7 with prettier integration

### ISSUE THREAD

**Issue by Validark** [typescript-eslint/typescript-eslint#1151]
For some reason, my prettier-eslint is getting a parsing error. Is there an easy fix for this?

```ts
const a: { x: number; y: number; z?: number } = { x: 1, y: 2 };
const b = a.z ?? 3;
```

**What did you expect to happen?** Parse without errors

**What actually happened?**
![image](https://user-images.githubusercontent.com/15217173/67713414-c6a93f00-f993-11e9-904a-0a28d3edf9d3.png)

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.6.0` |
| `TypeScript`                | `3.7.1-rc` |
| `ESLint`                    | `6.6.0` |
| `node`                      | `12.9.1` |
| `npm`                       | `6.11.2` |

Here is my devDependencies in package.json:
```json
{
    "devDependencies": {
        "@typescript-eslint/eslint-plugin": "^2.6.0",
        "@typescript-eslint/parser": "^2.6.0",
        "eslint": "^6.6.0",
        "eslint-config-prettier": "^6.5.0",
        "eslint-plugin-prettier": "^3.1.1",
        "eslint-plugin-roblox-ts": "file:../../../Documents/GitHub/lint-rules",
        "typescript": "^3.7.1-rc"
    },
}
```
And my .eslintrc
```json
{
	"parser": "@typescript-eslint/parser",
	"parserOptions": {
		"jsx": true,
		"useJSXTextNode": true,
		"ecmaVersion": 2020,
		"sourceType": "module",
		"project": "./tsconfig.json"
	},
	"plugins": ["@typescript-eslint", "@typescript-eslint/eslint-plugin", "prettier"],
	"extends": [
		"plugin:@typescript-eslint/recommended",
		"prettier/@typescript-eslint",
		"plugin:prettier/recommended",
	],
	"rules": {
		"prettier/prettier": [
			"warn",
			{
				"semi": true,
				"trailingComma": "all",
				"singleQuote": false,
				"printWidth": 120,
				"tabWidth": 4,
				"useTabs": true
			}
		],
	}
}
```

**Comment by bradzacher** [comment]
https://github.com/prettier/prettier/pull/6657#event-2749290110

They only just merged support for 3.7 features (using the latest version of our parser).
It has not been released yet. Until then, prettier will error!

**Comment by EdenTurgeman** [comment]
@bradzacher 3.7 is officially released, will prettier still error?

**Comment by bradzacher** [comment]
It will until prettier/prettier#6657 is released (I think the next version is 1.19.0)
Keep an eye out for prettier's next release.
